### ISSUE TITLE
parse_json_array() broken in 0.3.7

### ISSUE THREAD

**Issue by predictiple** [Velocidex/velociraptor#198]
It seems to be this that broke it: [0193176](https://github.com/Velocidex/velociraptor/commit/019327603cc969d60734497182525d7895bd01cf#diff-00f395ac9ddf0c99511ec82f53eedb0c)

```
velociraptor-v0.3.7-windows-amd64.exe artifacts collect -v Generic.Forensic.Carving.URLs
[INFO] 2019-12-09T08:31:14Z Loaded 137 built in artifacts
[][][INFO] 2019-12-09T08:31:14Z parse_json_array: expect JSON object open with '{'
[][INFO] 2019-12-09T08:31:14Z Collection completed in 0 Seconds
```

With 0.3.6 no problem.

**Comment by scudette** [comment]
0.3.7 made parse_json_array more strict by requiring the array to contain
dicts. Can you check what kind of array it is trying to parse?

The reason being that we now maintain key ordering when parsing json.

**Comment by scudette** [comment]
I actually think it is a useful feature to support array of all types. Fixed in #200

**Comment by predictiple** [comment]
Thanks for attending to this so quickly. I've tested with the CI build and all is good.
